sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/OuterJoinEliminationSuite.scala

```diff
@@ -268,4 +268,59 @@ class OuterJoinEliminationSuite extends PlanTest {
 
     comparePlans(optimized, originalQuery.analyze)
   }
+
+  test("SPARK-39172: Remove left outer join if only left-side columns being selected and " +
```

Comment:
Remove left/right outer join if only left/right side columns are selected and the join keys on the other side are unique.

Reply:
The same to PR title

Reply:
The PR title can be `Remove left/right outer join if only left/right side columns are selected and the join keys on the other side are unique`